sdk/main/include/impl/HttpClient.h

```diff
@@ -0,0 +1,58 @@
+#include <curl/curl.h>
+#include <iostream>
+#include <string>
+#include <memory>
+
+class HttpClient
+{
+public:
+  /**
+   * @brief Constructor for HttpClient.
+   *
+   * This constructor initializes the HttpClient and performs global libcurl initialization
+   * using CURL_GLOBAL_DEFAULT.
+   *
+   * @throw std::runtime_error If libcurl initialization fails.
+   */
+  HttpClient();
+
+  HttpClient(const HttpClient&) = delete;            // no need as only one httpClient could be used
+  HttpClient& operator=(const HttpClient&) = delete; // no need as only one httpClient could be used
```

Comment:
If only one `HttpClient` can/should be used, should this class be made a singleton?

Reply:
Bad comment explanation on my side. We can have several httpClients, but I could not think of a use case in which we need to copy them as they don`t have specific states to be copied and the operations are then useless.